src/sidebar/util/test/thread-test.js

```diff
@@ -0,0 +1,36 @@
+import * as threadUtil from '../thread';
+
+describe('sidebar/util/thread', () => {
+  let fakeThread = {
+    annotation: {},
+    visible: true,
+    children: [
+      {
+        annotation: {},
+        visible: true,
+        children: [
+          { annotation: {}, visible: true, children: [] },
+          { annotation: {}, visible: false, children: [] },
+        ],
+      },
+      {
+        annotation: {},
+        visible: false,
+        children: [{ annotation: {}, visible: true, children: [] }],
+      },
+      { annotation: {}, visible: true, children: [] },
+    ],
+  };
+
+  describe('visibleCount', () => {
+    it('should count the number of visible entries in the thread', () => {
+      assert.equal(threadUtil.countVisible(fakeThread), 5);
```

Comment:
I think it would be a good idea here to have fixtures which cover the case where the input thread is hidden and the case where it is visible, since there are branches for both of those. Same with tests for `countHidden`.